toucan_connectors/google_big_query/google_big_query_connector.py

```diff
@@ -236,6 +236,39 @@ def _build_dataset_info_query(dataset_ids: Iterable[str]) -> str:
             for dataset_id in dataset_ids
         )
 
+    def _get_project_structure_fast(
+        self, client: bigquery.Client, dataset_ids: Iterable[str]
+    ) -> pd.DataFrame:
+        """Retrieves the project's structure in a single query.
+
+        Only works if all datasets are in the same location.
+        """
+        query = self._build_dataset_info_query(dataset_ids)
+        return client.query(query).to_dataframe()
+
+    def _get_project_structure_slow(
+        self, client: bigquery.Client, dataset_ids: Iterable[str]
+    ) -> pd.DataFrame:
+        """Retrieves the project's structure in a multiple query.
+
+        Works even if the project's datasets are in different locations.
+        """
+        # In case the previous query failed, we need to get information for every dataset in the
```

Comment:
```suggestion
        # In case datasets are not in the same location, we need to get information for every dataset in the
```
Or we could even remove this comment now that there is a nice docstring

Reply:
I've left it in order to explain why we need to re-fetch information about every single dataset